src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -1,207 +1,375 @@
+                    {
+                        if (_receiver == null || cancellationToken.IsCancellationRequested)
+                        {
+                            return;
+                        }
+
+                        if (_isSessionsEnabled)
+                        {
+                            try
+                            {
+                                receiver = await sessionClient.AcceptMessageSessionAsync();
+                            }
+                            catch (ServiceBusTimeoutException)
+                            {
+                                // it's expected if the entity is empty, try next time
+                                continue;
+                            }
+                        }
+
+                        IList<Message> messages = messages = await receiver.ReceiveAsync(_serviceBusOptions.BatchOptions.MaxMessageCount, _serviceBusOptions.BatchOptions.OperationTimeout);
+
+                        if (messages != null)
+                        {
+                            Message[] messagesArray = messages.ToArray();
+                            ServiceBusTriggerInput input = new ServiceBusTriggerInput()
+                            {
+                                Messages = messagesArray
+                            };
+                            if (_isSessionsEnabled)
+                            {
+                                input.MessageSession = receiver as IMessageSession;
+                            }
+                            else
+                            {
+                                input.MessageReceiver = receiver as MessageReceiver;
+                            }
+
+                            FunctionResult result = await _triggerExecutor.TryExecuteAsync(GetTriggerFunctionData(messagesArray, input), cancellationToken);
+
+                            if (cancellationToken.IsCancellationRequested)
+                            {
+                                return;
+                            }
+
+                            // Complete batch of messages only if the execution was succesfull
+                            if (result.Succeeded && Receiver != null)
+                            {
+                                await receiver.CompleteAsync(messagesArray.Select(x => x.SystemProperties.LockToken));
+                            }
+                        }
+                    }
+                    catch (ObjectDisposedException)
+                    {
+                        // Ignore as we are stopping the host
+                    }
+                    catch (NullReferenceException)
+                    {
+                        // Ignore as we are stopping the host
+                    }
+                    catch (Exception ex)
+                    {
+                        // Log another exception
+                        _logger.LogError($"Exception has occurred during batch listening:: {ex.ToString()}");

```

Comment:
nit: "An unhandled exception occurred in the message batch receive loop: {ex.ToString()}"

Reply:
What if an exception occurs that is fatal, meaning listening is no longer happening? Can that happen? Wondering if this listener can get in to a zombie failed state.

Reply:
The listener should not go to "zombie  failed  state". Listening is happen on each loop step. I will monitor unhandled exceptions after the feature release.